app/models/host/managed.rb

```diff
@@ -684,12 +684,12 @@ def provider
   def root_pass
     return self[:root_pass] if self[:root_pass].present?
     return hostgroup.try(:root_pass) if hostgroup.try(:root_pass).present?
-    Setting[:root_pass]
+    crypt_root_pass(Setting[:root_pass])&.first
```

Comment:
I do not have time for full review, but what immediately strikes me is that previously this was just a getter, now this modifies state of the model. In the method both `root_pass=` and `grub_pass=` methods are called.